pkg/config/visibility/visibility.go

```diff
@@ -14,24 +14,36 @@
 
 package visibility
 
-import "fmt"
+import (
+	"fmt"
 
-// Instance defines whether a given config or service is exported to local namespace, all namespaces or none
+	"istio.io/istio/pkg/config/labels"
+)
+
+// Instance defines whether a given config or service is exported to local namespace, some set of namespaces, or
+// all namespaces or none
 type Instance string
 
 const (
 	// Private implies namespace local config
 	Private Instance = "."
 	// Public implies config is visible to all
 	Public Instance = "*"
+	// None implies service is visible to no one. Used for services only
+	None Instance = "~"
 )
 
-// Validate a visibility value.
+// Validate a visibility value ( ./*/~/some namespace name which is DNS1123 label)
 func (v Instance) Validate() (errs error) {
 	switch v {
 	case Private, Public:
 		return nil
+	case None:
+		return fmt.Errorf("exportTo ~ (none) is not allowed for Istio configuration objects")
 	default:
-		return fmt.Errorf("only . or * is allowed in the exportTo in the current release")
+		if !labels.IsDNS1123Label(string(v)) {
```

Comment:
Actually, this is not right. The allowed length is 63 here. But in k8s the allowed name length is <= 253

Reply:
I have found many misuse of `IsDNS1123Label` in the whole repo

Reply:
Are you sure kubernetes namespaces can be >63 characters? If the name is 253 characters, then no one can create a service inside a namespace as the total DNS name length will be >255 characters.

Reply:
You are right, i am looking at something wrong